platforms/documentation/docs/src/docs/userguide/dep-man/basics/variant_aware_resolution.adoc

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[variant_aware_resolution]]
+= 7. Variant Aware Dependency Resolution
+
+In Gradle, the traditional method of sharing artifacts between projects involves the producer defining a configuration, and the consumer explicitly stating which configuration to depend on.
+However, this method isn't ideal when you want to automate the selection of *variants* based on the consumer's needs without specifying a particular configuration.
+
+Gradle allows for a more dynamic approach called **variant-aware resolution**, where both the producer and consumer define their requirements using **attributes**.
+
+By doing this, Gradle can automatically select the correct *variant* without the consumer explicitly specifying which one to use.
+
+For instance, if you're working with different architectures (like `arm64` and `i386`), Gradle can choose the appropriate version of a library (`myLib`) for each architecture:
+
+1. The *producer*, `myLib`, sets up configurations (`arm64Jars`, `i386Jars`) with specific *attributes* (e.g., `ArchType.ARM64`, `ArchType.I386`).
+2. The *consumer*, `myApp`, specifies the required *attributes* (e.g., `ArchType.ARM64`) in its configuration (`testRuntimeClasspath`).
+3. If the *consumer*, `myApp`, needs `arm64` architecture, Gradle will automatically pick the `arm64Jars` *variant* from the `myLib` *producer* and use the corresponding *artifact*.
+
+== A coded example
+
+Consider a Java library where you create a new *variant* called `instrumentedJars` and want to ensure it’s selected for testing:
+
+1. *Producer Project*: Creates a specialized `instrumentedJars` *variant* marked with specific attributes.
+2. *Consumer Project*: Configured to request the `instrumented-jar` *variant* for testing.
+3. *Compatibility Rule*: Ensures that if the `instrumented-jar` isn’t available, Gradle falls back on the regular `jar`.
+
+Let's look at the build files of the producer and consumer.
+
+== The producer side
+
+=== 1. Create an instrumented JAR:
+
+Our Java library has a task called `instrumentedJar` which produces a JAR file with the classifier "instrumented."
```

Comment:
```suggestion
Our Java library has a task called `instrumentedJar` which produces a JAR file.
```

"Classifier" is a maven term and we should probably avoid it here.

Reply:
Done.